src/remote/configuration/toml.py

```diff
@@ -203,20 +215,51 @@ def _merge_field(field: str, global_config: GlobalConfig, local_config: LocalCon
     return result
 
 
-def _get_exclude(sync_rules: Optional[SyncRulesConfig]) -> List[str]:
+def _get_exclude(sync_rules: Optional[SyncRulesConfig], workspace_root: Path) -> List[str]:
     if sync_rules is None:
         return []
-    return sync_rules.exclude
+    exclude = sync_rules.exclude
+
+    gitignore = workspace_root / ".gitignore"
+    if sync_rules.include_vsc_ignore_patterns and gitignore.exists():
+        with gitignore.open() as f:
+            for line in f.readlines():
+                line = line.strip()
+                if not line or line.startswith("#"):
+                    continue
+
+                exclude.append(line)
+    return exclude
+
+
+def _clean_up_dict(data_dict: Dict[str, Any]) -> None:
+    for key in data_dict:
+        if isinstance(data_dict[key], Path):
+            data_dict[key] = str(data_dict[key])
+        elif isinstance(data_dict[key], dict):
+            _clean_up_dict(data_dict[key])
+        elif isinstance(data_dict[key], (list, tuple)):
+            for item in data_dict[key]:
+                if isinstance(item, dict):
+                    _clean_up_dict(item)
 
 
-def _merge_excludes(
```

Comment:
When do you expect to be saving the config file?
Is this an auto-migration path?

Reply:
We save config files when somebody runs commands like `remote-init`, `remote-add` or `remote-ignore`. 

Also, I don't think that any automigration is nessesary, we will just do this type of config default soon and at some point old-style codebases will migrate by themselves